src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24414.cs

```diff
@@ -16,8 +17,76 @@ public Issue24414(TestDevice device)
 		[Category(UITestCategories.Visual)]
 		public void Issue24414Test()
 		{
-			App.WaitForElement("WaitForStubControl");
-			VerifyScreenshot();
+			// TODO: Refactor after collecting new screenshot from AZP
+			// - TryVerifyScreenshot => VerifyScreenshot
+			// - Remove duplicated call to `VerifyMemory`
+
+			var label = App.WaitForElement("TheLabel");
+			var ex = TryVerifyScreenshot("Issue24414Test");
+#if ANDROID
+			var memorySnapshot = App.GetPerformanceMemoryInfo();
+#endif
+
+			for (int i = 1; i <= 5; i++)
+			{
+				label.Tap();
+				var exception = TryVerifyScreenshot("Issue24414Test_" + i);
+				ex ??= exception;
+			}
+
+			if (ex != null)
+			{
+#if ANDROID
+				VerifyMemory(memorySnapshot, App.GetPerformanceMemoryInfo());
+#endif
+
+				throw ex;
+			}
+
+#if ANDROID
+			VerifyMemory(memorySnapshot, App.GetPerformanceMemoryInfo());
+#endif
+		}
+
+#if ANDROID
+		static void VerifyMemory(IReadOnlyDictionary<string, int> memoryBefore, IReadOnlyDictionary<string, int> memoryAfter)
+		{
+			var allowedMemoryIncrease = new Dictionary<string, double>
```

Comment:
@jonathanpeppers any suggestions on how to set up this properly?
The test is failing on CI.. maybe I should avoid checking some of them.. what do you think?